### ISSUE TITLE
Code with `` and equation $$ can't work

### ISSUE THREAD

**Issue by LeenHawk** [BryceRussell/astro-remark-description#2]
It seems description delete them automatically, how should I set the option to turn it off?

**Comment by LeenHawk** [comment]
I have used remark and rehype to do it by myself, but it's worth some options in my view as description may be use rendered in abstract in some article list card.

**Comment by LeenHawk** [comment]
In fact, I find a easier way to get description for me. We can get full fresh markdown string from `post.body`. So if you are migrated from hexo or hugo, just get substring before`<!-- more -->`.

**Comment by BryceRussell** [comment]
Great catch @LeenHawk ! This is definitely something I want to add support for, currently [I use  `mdast-to-string`](https://github.com/BryceRussell/astro-remark-description/blob/master/index.js#L29C18-L29C18) to get the text of a node which sanitizes all HTML/markdown from the string.

What should the API for this look like? Currently I am thinking of :

1) Exposing the mdast node to the `transform` function

```js
import { defineConfig } from 'astro/config';
import { toHast } from 'mdast-util-to-hast'
import { toHtml } from "hast-util-to-html";

export default defineConfig({
  markdown: {
    remarkPlugins: [
      [remarkDescription, { 
        transform: (text, { node }) => toHtml(toHast(node))
      }]
    ]
  }
});
```
This option gives you complete freedom over how the node is transform into text, but it is a bit more advanced to setup and you have to parse the mdast to html yourself

2) Add a `html` option to toggle between text and HTML

```js
import { defineConfig } from 'astro/config';

export default defineConfig({
  markdown: {
    remarkPlugins: [
      [remarkDescription, { 
        html: true
      }]
    ]
  }
});
```
This option is a lot simpler, but it is less flexible than the first and would require more dependencies by default

**Comment by LeenHawk** [comment]
I think I don't understand what do you mean. Could you parse string use astro remark plugins and options or set it in astro config file? Or it's mean we need to use mdast to get code and equations support by option? Could we use a lazier way to make it?

For someone migrated from hexo or hugo like me, we have made `<!-- more -->` in all post, so we just get the substring before this label and rendered it. You should focus those need to more options.

It's my code for this:

```js
import { type CollectionEntry } from "astro:content";
import {unified} from 'unified'
import remarkParse from 'remark-parse'
import remarkMath from 'remark-math'
import remarkRehype from 'remark-rehype'
import rehypeMathjax from 'rehype-mathjax'
import rehypeStringify from 'rehype-stringify'
function outputBeforeMore(str) {
  const index = str.search(/<!--\s*more\s*-->/i);
  if (index !== -1) {
    return str.substring(0, index);
  }
  return str;
}
async function getdescription(post: CollectionEntry<"posts">) {
  const descriptionraw:string = outputBeforeMore(post.body);
  const description=await unified()
  .use(remarkParse)
  .use(remarkMath)
  .use(remarkRehype)
  .use(rehypeMathjax)
  .use(rehypeStringify)
  .process(descriptionraw)
  return String(description);
}
```
The only inconvenient thing is I don't find astro render string API, the remote markdown seems to be dealt with importing all package in frontmatter by myself.

**Comment by LeenHawk** [comment]
In fact, a more tricky but feasible way used in hugo is you only parse automatically when user don't add `<!--more-->` label, and it works well.

**Comment by BryceRussell** [comment]
Ahh, interesting! You can use the `node` option with `remarkDirective` to emulate this use-case, for example:

```js
import { defineConfig } from 'astro/config';
import remarkDescription from 'astro-remark-description'
import remarkDirective from 'remark-directive'

export default defineConfig({
  markdown: {
    remarkPlugins: [
      remarkDirective,
      [remarkDescription, { 
        node: node => node.type === 'containerDirective' && node.name === 'description'
      }]
    ]
  }
});
```

```md
# Heading

Some text

:::description
Some more text
:::
```

I should add an example for this in the README

For the option to render the node to HTML I think I am going to add both options I mentioned, that way you can easily parse it to HTML using the `html` option or do something completely custom using the `transform` function

**Comment by BryceRussell** [comment]
I played around with this a little more and I was able to get the to work with the `<!-- more -->` comment using a configuration like this: 

```js
import { defineConfig } from 'astro/config';
import remarkDescription from 'astro-remark-description'

// https://astro.build/config
export default defineConfig({
  markdown: {
    remarkPlugins: [
      [remarkDescription, { 
        node: (node, i, parent) => {
          const sibling = parent?.children[i + 1]
          return sibling?.type === 'html' && sibling?.value === '<!-- more -->'
        }
      }]
    ]
  }
});

```

**Comment by LeenHawk** [comment]
> I played around with this a little more and I was able to get the to work with the `<!-- more -->` comment using a configuration like this:
> 
> ```js
> import { defineConfig } from 'astro/config';
> import remarkDescription from 'astro-remark-description'
> 
> // https://astro.build/config
> export default defineConfig({
>   markdown: {
>     remarkPlugins: [
>       [remarkDescription, { 
>         node: (node, i, parent) => {
>           const sibling = parent?.children[i + 1]
>           return sibling?.type === 'html' && sibling?.value === ''
>         }
>       }]
>     ]
>   }
> });
> ```

Fantastic work, thank you.

**Comment by BryceRussell** [comment]
I am going to keep this issue open until I create a PR to add HTML rendering as a feature

**Comment by BryceRussell** [comment]
Hey @LeenHawk, I just published `v1.1.1` which adds support for [rendering a node to HTML](https://github.com/BryceRussell/astro-remark-description#html) and added documentation for [emulating Hexo](https://github.com/BryceRussell/astro-remark-description#using-remarkdirective-to-target-a-node).

You can upgrade to the new version using the command `npm update astro-remark-description`

Thanks for raising this as an issue! ❤️
